plugins/processors/noise/README.md

```diff
@@ -0,0 +1,63 @@
+# Noise Processor
+
+The *Noise* processor is used to add noise to numerical field values. For each field a noise is generated using a defined probability densitiy function and added to the value. The function type can be configured as _Laplace_, _Gaussian_ or _Uniform_.
+Depending on the function, various parameters need to be configured:
+
+## Configuration
+
+Depending on the choice of the distribution function, the respective parameters must be set. Default settings are `noise_type = "laplacian"` with `mu = 0.0` and `scale = 1.0`:
+
+```toml
+[[processors.noise]]
+  scale = 1.0
+  mu = 0.0
+  noise_type = "laplacian"
+```
+
+Using the `include_fields` and `exclude_fields` options a filter can be configured to apply noise only to numeric fields matching it.
+The following distribution functions are available.
+
+### Laplacian
+
+* `noise_type = laplacian`
+* `scale`: also referred to as _diversity_ parameter, regulates the width & height of the function, a bigger `scale` value means a higher probability of larger noise, default set to 1.0
+* `mu`: location of the curve, default set to 0.0
+
+### Gaussian
+
+* `noise_type = gaussian`
+* `mu`: mean value, default set to 0.0
+* `scale`: standard deviation, default set to 1.0
+
+### Uniform
+
+* `noise_type = uniform`
+* `min`: minimal interval value, default set to -1.0
+* `max`: maximal interval value, default set to 1.0
+
+## Example
+
+Add noise to each value the *Inputs.CPU*  plugin generates, except for the _usage\_steal_, _usage\_user_, _uptime\_format_, _usage\_idle_ field and all fields of the metrics _swap_, _disk_ and _net_:
+
+```toml
+[[inputs.cpu]]
+  percpu = true
+  totalcpu = true
+  collect_cpu_time = false
+  report_active = false
+
+[[processors.noise]]
+  scale = 1.0
+  mu = 0.0
+  noise_type = "laplacian"
+  include_fields = []
+  exclude_fields = ["usage_steal", "usage_user", "uptime_format", "usage_idle" ]
```

Comment:
I think `exclude_fields` has the same effect as the existing setting `namedrop`. See https://github.com/influxdata/telegraf/blob/master/docs/CONFIGURATION.md#selectors "If a metric is excluded from a Processor or Aggregator plugin, it is skips the plugin and is sent onwards to the next stage of processing."

Similarly, `include_fields` seems to do the same thing as `namepass`.

Reply:
I just tested it, but the options ``namedrop`` or ``fielddrop`` either completely drop the whole metric or discard the field. But ``exclude_fields`` can be used, if the field should be emitted _without_ any noise modifications to it.

Reply:
@reimda the idea here is to skip certain fields from adding noise and leave them in their original state, so I think this cannot be achieved using the usual telegraf tools.

Reply:
There is a case in the running processor where the metric is passed straight to the accumulator in their original state without going through the processor. I think this is what you're trying to do with include_fields.

https://github.com/influxdata/telegraf/blob/f0f4a055e96692ef6786f72d7ee363b8a158da37/models/running_processor.go#L84

Reply:
@reimda I don't think so. Imagine a metric with numeric fields `A`, `B` and `C`. Now we want to only add noise to field `B` and leave the others in their original state (without noise added). In this case we want the metric to be processed but only change field `B`. This can be done by setting `include_fields = ["B"]` or by `exclude_fields = ["A", "C"]` (which is for convenience and for cases where the other field names are not known).
For any case the input as well as the output should have the fields `A`, `B` and `C` in our example.

I cannot see how to do it with the telegraf options as they will either prevent the whole metric to be processed or drop fields from that metric, aren't they?

Reply:
I see what you're saying now. These act on fields in a metric while namepass/namedrop act on the whole metric. Thanks for the example. I'm fine with adding these new settings.